elements/lisk-dpos/src/dpos.ts

```diff
@@ -208,4 +208,34 @@ export class Dpos {
 			delegateListRoundOffset,
 		});
 	}
+
+	// tslint:disable-next-line: prefer-function-over-method
+	public async isDPoSProtocolCompliant(
+		blockHeader: BlockHeader,
+		store: StateStore,
+	): Promise<boolean> {
+		const lastBlockHeaders = store.consensus.lastBlockHeaders;
+		const delegateForgedBlocks = lastBlockHeaders.filter(
+			block => block.generatorPublicKey === blockHeader.generatorPublicKey,
+		);
+
+		if (!delegateForgedBlocks.length) {
+			// If the forger din't forge any block in the last three rounds
+			return true;
+		}
+
+		const { seedReveal: previousBlockSeedReveal } = delegateForgedBlocks.slice(
+			0,
+			1,
+		)[0];
+		const { seedReveal: newBlockSeedReveal } = blockHeader;
+
+		// Check if last block seedReveal is not a preimage of new block
```

Comment:
I think it should be opposite if I understood it correctly,
`// Check if the new block seed reveal is not a preimage of the last block seed reveal`

Reply:
also, I think it sounds like a correct scenario if it's not equal, maybe emphasis on what it should be,
`New block seed reveal should be a preimage of the last block seed reveal`